src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java

```diff
@@ -0,0 +1,636 @@
+            return pColums.length;
+        }
+
+		@Override
+        public Object getValueAt(int r, int c) {
+            MedicalInventory medInvt = inventoryList.get(r);
+
+            if (c == -1) {
+                return medInvt;
+            } else if (c == 0) {
+                return medInvt.getInventoryReference();
+            } else if (c == 1) {
+                Ward ward = new Ward();
+                try {
+                    ward = wardBrowserManager.findWard(medInvt.getWard());
+                } catch (OHServiceException e) {
+                    OHServiceExceptionUtil.showMessages(e);
+                }
+                return ward == null ? "" : ward.getDescription();
+            } else if (c == 2) {
+                return medInvt.getInventoryDate().format(DATE_TIME_FORMATTER);
+            } else if (c == 3) {
+                return medInvt.getStatus();
+            } else if (c == 4) {
+                return medInvt.getUser();
+            }
+            return null;
+        }
+
+        @Override
+        public boolean isCellEditable(int arg0, int arg1) {
+            return false;
+        }
+
+    }
+
+    private JComboBox<String> getComboBox() {
+        if (statusComboBox == null) {
+            statusComboBox = new JComboBox<>();
+            statusComboBox.addItem("");
+            for (InventoryStatus currentStatus : InventoryStatus.values()) {
+                statusComboBox.addItem(MessageBundle.getMessage("angal.inventory." + currentStatus));
+            }
+			statusComboBox.addActionListener(actionEvent -> {
+				InventoryBrowsingModel inventoryBrowsingModel = new InventoryBrowsingModel();
+				totalRows = inventoryBrowsingModel.getRowCount();
+				startIndex = 0;
+				previous.setEnabled(false);
+				next.setEnabled(totalRows > PAGE_SIZE);
+				jTableInventory.setModel(new InventoryBrowsingModel(startIndex, PAGE_SIZE));
+				initialiseCombo(totalRows);
+			});
+        }
+        return statusComboBox;
+    }
+
+    private JLabel getStatusLabel() {
+        if (statusLabel == null) {
+            statusLabel = new JLabel(MessageBundle.getMessage("angal.inventory.status.label"));
+            statusLabel.setHorizontalAlignment(SwingConstants.RIGHT);
+        }
+        return statusLabel;
+    }
+
+    private void ajustWidth() {
+		for (int i = 0; i < pColumwidth.length; i++) {
+			jTableInventory.getColumnModel().getColumn(i).setMinWidth(pColumwidth[i]);
+		}
+    }
+
+    public void initialiseCombo(int total_rows) {
+		int j = 0;
+
+		pagesCombo.removeAllItems();
+		for (int i = 0; i < total_rows / PAGE_SIZE; i++) {
+			j = i + 1;
+			pagesCombo.addItem(j);
+		}
+		if (j * PAGE_SIZE < total_rows) {
+			pagesCombo.addItem(j + 1);
+			under.setText("/" + (total_rows / PAGE_SIZE + 1 + " " + MessageBundle.getMessage("angal.inventory.pages.text")));
+		} else {
+			under.setText("/" + total_rows / PAGE_SIZE + " " + MessageBundle.getMessage("angal.inventory.pages.text"));
```

Comment:
isn't there a formatted msg in the bundle that can by used instead of concatentation?

Reply:
not in this PR, but in #2125. We can add it in both and then resolve conflicts, or first merge the latter...